### PR TITLE
Case 20301 - add platform info

### PR DESCRIPTION
**Pull request by NissimHadar**
https://highfidelity.fogbugz.com/f/cases/20301/Design-and-implement-a-way-to-run-test-ONLY-on-specific-Platform-Profile
# Description
Adds a number of methods to read platform information.  This will be used to enable tests that are platform-specific.
The methods are available from scripts, allowing script writers to check the capabilities of the platform Interface is running on.
# Testing
Run the following tests on both Windows and Mac
1.  Start Interface
2.  Open the JS console
3.  Enter `PlatformInfo.getOperatingSystemType()`
     Verify correct operating system is returned
4.  Enter `PlatformInfo.getCPUBrand()`
     Verify correct CPU brand is returned
5.  Enter `PlatformInfo.getNumLogicalCores()`
     Verify correct number of logical cores is returned
6.  Enter `PlatformInfo.getTotalSystemMemoryMB()`
     Verify that the correct total memory size is returned (in mebibytes)
7.  Enter `PlatformInfo.getGraphicsCardType()`
     Verify that the correct graphics card type is returned.

On Windows with Rift:
1.  Enter `PlatformInfo.hasRiftControllers()`
     Verify that true is returned
2.  Enter `PlatformInfo.hasViveControllers()`
     Verify that false is returned

On Windows with Vive:
1.  Enter `PlatformInfo.hasRiftControllers()`
     Verify that false is returned
2.  Enter `PlatformInfo.hasViveControllers()`
     Verify that true is returned

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16180/HighFidelity-Beta-PR14650-c0a784f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16180/HighFidelity-Beta-PR14650-c0a784f.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16180/HighFidelity-Beta-Interface-PR14650-c0a784f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16180/HighFidelity-Beta-PR14650-c0a784f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16180/HighFidelity-Beta-Interface-PR14650-c0a784f.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16192/HighFidelity-Beta-PR14650-56ee91d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16192/HighFidelity-Beta-PR14650-56ee91d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16192/HighFidelity-Beta-Interface-PR14650-56ee91d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16192/HighFidelity-Beta-PR14650-56ee91d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16192/HighFidelity-Beta-Interface-PR14650-56ee91d.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16202/HighFidelity-Beta-PR14650-65d975f.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16202/HighFidelity-Beta-PR14650-65d975f.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16202/HighFidelity-Beta-Interface-PR14650-65d975f.exe

File: interface/src/scripting/PlatformInfoScriptingInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+QString PlatformInfoScriptingInterface::getOperatingSystemType() {
+#ifdef Q_OS_WIN
+    return "WINDOWS";
+#elif defined Q_OS_MAC
+    return "MACOS";
+#else
+    return "UNKNOWN";
+#endif
```
Comment (sethalves) [inline interface/src/scripting/PlatformInfoScriptingInterface.cpp]: why no Q_OS_ANDROID or Q_OS_LINUX ?

**Comment by qliejahner** [comment]
Currently testing

File: tools/nitpick/src/TestRunner.cpp [hunk 2]
```diff
@@ -469,12 +469,28 @@ void TestRunner::runInterfaceWithTestScript() {
         url = "hifi://localhost";
     }
 
+    QString deleteScript =
+        QString("https://raw.githubusercontent.com/") + _user + "/hifi_tests/" + _branch + "/tests/utils/deleteNearbyEntities.js";
```
Comment (sethalves) [inline tools/nitpick/src/TestRunner.cpp]: QUrl might make this more robust

File: tools/nitpick/src/ui/Nitpick.cpp [hunk 3]
```diff
@@ -287,7 +286,7 @@ void Nitpick::about() {
 }
 
 void Nitpick::content() {
-    _helpWindow.show();
+    QDesktopServices::openUrl(QUrl("https://github.com/highfidelity/hifi/blob/master/tools/nitpick/README.md"));
```
Comment (sethalves) [inline tools/nitpick/src/ui/Nitpick.cpp]: should this URL be in a named constant?

File: interface/src/scripting/PlatformInfoScriptingInterface.cpp (hunk 1 above)
Comment (NissimHadar) [reply interface/src/scripting/PlatformInfoScriptingInterface.cpp]: We don't (yet) run tests on either Android or Linux.  We don't have a Linux build, and we cannot run nitpick on Android.  I did add those at the beginning, but I couldn't test them, and I don't feel comfortable implementing something I haven't fully tested.

**Comment by qliejahner** [comment]
PASS (if changes are not made in response to seth's comments)

Tested with Rift and Vive with Nvidia cards, a desktop with AMD, and Mac. All commands returned expected values on each machine.

File: tools/nitpick/src/TestRunner.cpp (hunk 2 above)
Comment (NissimHadar) [reply tools/nitpick/src/TestRunner.cpp]: Thanks - I will look into it, as this usage pattern is common.

File: tools/nitpick/src/ui/Nitpick.cpp (hunk 3 above)
Comment (NissimHadar) [reply tools/nitpick/src/ui/Nitpick.cpp]: Actually the user (highfidelity) and branch should be variables - I will fix that.

**Pull request merged**